### ISSUE TITLE
Compilation fails when defining mapper as inner/nested class and the outer class has a parent class

### ISSUE THREAD

**Issue by renatosc** [riok/mapperly#548]
**Describe the bug**
When defining the mapper as an inner class, the generated mapper code is copying the existing parent class which is not needed. Normally this wouldn't be issue if your parent class is in the same namespace, but if it is on a different namespace you will get an compilation error since currently Mapperly doesn't support copying the "using namespaces" of the file. 

**To Reproduce**
1. Create a class with an attribute
2. Declare a mapper as an inner class and have the outer class inheriting from a class from a different namespace
3. Try to compile the code.

**Expected behavior**
Code compiles without error

**Code snippets**
```csharp
using myfactory;
using Riok.Mapperly.Abstractions;

namespace Bug_Mapperly
{

    internal class Program { static void Main(string[] args) { Console.WriteLine("Hello, World!"); } }


    public partial class CarDto: VehicleDto
    {
        public string? Maker { get; set; }
        public string? Model { get; set; }

        [Mapper]
        private partial class CarMapper
        {
            public partial CarDto From(Car car);
        }

    }

    public class Car
    {
      public string? Maker { get; set; }
        public string? Model { get; set; }
    }

}

namespace myfactory
{
    public class VehicleDto
    {
        public string? Color { get; set; }
    }
}
```

**Fix suggestion**
Make the generated code to not copy the parent class inheritance

**Environment (please complete the following information):**
- Mapperly Version: 2.9.0-next.2
- .NET Version: .NET 6
- Target Framework: [e.g. .net6.0]
- IDE: Visual Studio Pro 2022
- OS: Windows 10

**Additional context**
A similar issue (copying attributes when not needed) was reported on https://github.com/riok/mapperly/issues/543#issuecomment-1625749856

**Comment by TimothyMakkison** [comment]
Thanks again. Should be fixed now 🤞, I've updated my original fix to strip the base class away.

**Comment by latonz** [comment]
Fixed by #544
